packages/fx-core/src/component/driver/teamsApp/teamsappMgr.ts

```diff
@@ -0,0 +1,325 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import {
+  Colors,
+  FxError,
+  Result,
+  TeamsAppInputs,
+  TeamsAppManifest,
+  err,
+  ok,
+} from "@microsoft/teamsfx-api";
+import AdmZip from "adm-zip";
+import fs from "fs-extra";
+import * as path from "path";
+import { Container } from "typedi";
+import * as util from "util";
+import { getLocalizedString } from "../../../common/localizeUtils";
+import { AppStudioScopes } from "../../../common/tools";
+import { FileNotFoundError, MissingRequiredInputError } from "../../../error/common";
+import { resolveString } from "../../configManager/lifecycle";
+import { createDriverContext } from "../../utils";
+import { envUtil } from "../../utils/envUtil";
+import { pathUtils } from "../../utils/pathUtils";
+import { DriverContext } from "../interface/commonArgs";
+import { ConfigureTeamsAppDriver, actionName as configureTeamsAppActionName } from "./configure";
+import { Constants } from "./constants";
+import {
+  CreateAppPackageDriver,
+  actionName as createAppPackageActionName,
+} from "./createAppPackage";
+import { ConfigureTeamsAppArgs } from "./interfaces/ConfigureTeamsAppArgs";
+import { CreateAppPackageArgs } from "./interfaces/CreateAppPackageArgs";
+import { PublishAppPackageArgs } from "./interfaces/PublishAppPackageArgs";
+import { ValidateAppPackageArgs } from "./interfaces/ValidateAppPackageArgs";
+import { ValidateManifestArgs } from "./interfaces/ValidateManifestArgs";
+import {
+  actionName as PublishAppPackageActionName,
+  PublishAppPackageDriver,
+} from "./publishAppPackage";
+import { manifestUtils } from "./utils/ManifestUtils";
+import { ValidateManifestDriver } from "./validate";
+import { ValidateAppPackageDriver } from "./validateAppPackage";
+
+class TeamsAppMgr {
+  async ensureAppPackageFile(inputs: TeamsAppInputs): Promise<Result<undefined, FxError>> {
+    // if no package file input, then do package first
+    if (!inputs["package-file"]) {
+      const packageRes = await this.packageTeamsApp(inputs);
+      if (packageRes.isErr()) {
+        return err(packageRes.error);
+      }
+      inputs["package-file"] = packageRes.value.outputZipPath;
+    }
+    if (!(await fs.pathExists(inputs["package-file"]))) {
+      return err(new FileNotFoundError("updateTeamsApp", inputs["package-file"]));
+    }
+    return ok(undefined);
+  }
+
+  async readManifestFromZip(appPackagePath: string): Promise<Result<TeamsAppManifest, FxError>> {
+    const archivedFile = await fs.readFile(appPackagePath);
```

Comment:
![image](https://github.com/OfficeDev/TeamsFx/assets/71362691/86cddb89-c475-430a-9ed5-9f160e99cfa7)

Can use this function to avoid duplicate.